lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -7563,7 +7563,7 @@ namespace Js
         {
            return res;
         }
-        if (JavascriptArray::Is(args[0]))
+        if (JavascriptArray::Is(args[0]) && !JavascriptArray::FromVar(args[0])->IsCrossSiteObject())
```

Comment:
Ok I guess it will just merge down. Anyway you'll want to retarget this branch to release/1.4 and rebase.